src/modalities/models/huggingface/collator.py

```diff
@@ -0,0 +1,226 @@
+    """
+
+    def __init__(
+        self,
+        sample_key: str,
+        target_key: str,
+        noise_density: float,
+        mean_noise_span_length: float,
+        tokenizer: TokenizerWrapper,
+    ):
+        self.sample_key = sample_key
+        self.target_key = target_key
+        self.noise_density = noise_density
+        self.mean_noise_span_length = mean_noise_span_length
+        self.tokenizer = tokenizer
+
+    def __call__(self, batch: List[Dict[str, torch.Tensor]]) -> DatasetBatch:
+        """The collator prepares data for an encoder-decoder model.
+        `samples` are used as encoder input, `targets` are shifted right by 1
+        and used as decoder input for autoregressive modeling with teacher forcing.
+        Length after masking is deterministic given noise_density and mean_noise_span_length:
+        all samples will have the same length, as will all targets.
+
+
+        Args:
+            batch (List[Dict[str, torch.Tensor]]): batch of input token ids
+
+        Returns:
+            DatasetBatch: Contains samples and targets with random spans masked out.
+        """
+        sample_tensor = torch.stack([torch.tensor(d[self.sample_key]) for d in batch])
+
+        batch_size, expanded_input_length = sample_tensor.shape
+        mask_indices = np.asarray([self.random_spans_noise_mask(expanded_input_length) for i in range(batch_size)])
+        labels_mask = ~mask_indices
+
+        input_ids_sentinel = self.create_sentinel_ids(mask_indices.astype(np.int8))
+        labels_sentinel = self.create_sentinel_ids(labels_mask.astype(np.int8))
+
+        input_ids = self.filter_input_ids(sample_tensor, input_ids_sentinel)
+        label_ids = self.filter_input_ids(sample_tensor, labels_sentinel)
+
+        samples = {self.sample_key: torch.from_numpy(input_ids)}
+        targets = {self.target_key: torch.from_numpy(label_ids)}
+
+        return DatasetBatch(targets=targets, samples=samples)
+
+    def create_sentinel_ids(self, mask_indices: np.ndarray[bool]) -> np.ndarray[int]:
+        """
+        Sentinel ids creation given the indices that should be masked.
+        The start indices of each mask are replaced by the sentinel ids in increasing
+        order. Consecutive mask indices to be deleted are replaced with `-1`.
+        Sentinel ids are defined by the tokenizer as 32099: <extra_id_0>, 32098: <extra_id_1>, ..., 32000: <extra_id_99>
+
+        Args:
+            mask_indices (np.ndarray[bool]): binary mask of tokens to mask out
+
+        Returns:
+            np.ndarray[int]: array of sentinel ids at starting point of spans to be masked out
+        """
+        start_indices = mask_indices - np.roll(mask_indices, 1, axis=-1) * mask_indices
+        start_indices[:, 0] = mask_indices[:, 0]
+
+        sentinel_ids = np.where(start_indices != 0, np.cumsum(start_indices, axis=-1), start_indices)
+        sentinel_ids = np.where(sentinel_ids != 0, (self.tokenizer.vocab_size - sentinel_ids), 0)
```

Comment:
we might want to replace the `tokenizer` parameter by the two parameters `eos_token_id` and `vocab_size`

Reply:
Done. eos_token_id is no longer needed